src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -147,9 +166,17 @@ public Keyspaces apply(Keyspaces schema)
         if (attrs.isCustom && attrs.customClass.equals(SASIIndex.class.getName()))
             return ImmutableSet.of(SASIIndex.USAGE_WARNING);
 
+        if (isDseIndexCreateStatement())
+            return ImmutableSet.of(String.format(DSE_INDEX_WARNING, indexName, attrs.customClass));
+
         return ImmutableSet.of();
     }
 
+    private boolean isDseIndexCreateStatement()
+    {
+        return DSE_INDEXES.contains(attrs.customClass);
```

Comment:
Just checking - case sensitivity isn't a concern here, right?

Reply:
Right. We expect a fully-qualified-name (case sensitive).